dmscripts/generate_user_email_list_applications.py

```diff
@@ -0,0 +1,110 @@
+import csv
+from functools import partial
+import time
+from multiprocessing.pool import ThreadPool
+
+from dmutils.apiclient.errors import HTTPError
+
+
+def find_suppliers(data_api_client):
+    for supplier in data_api_client.find_suppliers_iter():
+        yield supplier
+
+
+def execute_concurrently(callback, iterable_thing):
+    pool = ThreadPool(10)
+    return pool.imap_unordered(callback, iterable_thing)
+
+
+def filter_list_of_dicts_by_value(list_of_dicts, key_to_match, expected_value):
+    return [
+        dic for dic in list_of_dicts
+        if dic.get(key_to_match) == expected_value
+    ]
+
+
+def get_all_supplier_framework_info(data_api_client, framework_slug, suppliers):
+
+    def get_supplier_framework_info(enumerated_supplier):
+        """
+        Returns a supplier_framework or None if no supplier_framework entry exists
+        :param enumerated_supplier: tuple formatted (index, supplier)
+        :return: a supplier_framework or None
+        """
+        index, supplier = enumerated_supplier
+        if index % 100 == 0:
+            print('~{} suppliers'.format(index))
+        try:
+            supplier_framework = data_api_client.get_supplier_framework_info(supplier['id'], framework_slug)
+            return supplier_framework.get('frameworkInterest')
```

Comment:
Not for the route I'm calling: https://github.com/alphagov/digitalmarketplace-api/blob/master/app/main/views/suppliers.py#L324